core/queryalgebra/model/src/main/java/org/eclipse/rdf4j/query/algebra/Var.java

```diff
@@ -42,6 +45,11 @@ public Var(String name, Value value) {
 		setValue(value);
 	}
 
+	public Var(Value value) {
+		this(UUID.randomUUID() + "_anon");
```

Comment:
Nice idea to create an anonymous variable this way, but this will cause syntactically illegal variable names - in SPARQL, variable names cannot contain '-'. See createAnonVar in the SPARQL TupleExpBuilder, which use this to create the name: 
    
     new Var("_anon_" + UUID.randomUUID().toString().replaceAll("-", "_")); 

You probably also need to set the `isAnonymous` flag.

Might be a good idea to handle this in one place (here in the constructor) and refactor the TupleExprBuilder and other places where we create anonymous vars. We've had problems in the past with the naming pattern not being consistent, so DRYing this up is a good idea.

Reply:
I'm not sure if I use this code anymore, will have to take another look. Might be something left over from testing.